packages/middleware-endpoint/src/adaptors/getEndpointFromInstructions.ts

```diff
@@ -85,18 +85,42 @@ export const resolveParams = async <
 /**
  * Normalize some key of the client config to an async provider.
  * @private
+ *
+ * @param configKey - the key to look up in config.
+ * @param canonicalEndpointParamKey - this is the name the EndpointRuleSet uses.
+ *                                    it will most likely not contain the config
+ *                                    value, but we use it as a fallback.
+ * @param config - container of the config values.
+ * 
+ * @returns async function that will resolve with the value.
  */
-const createConfigProvider = <Config extends Record<string, unknown>>(
+export const createConfigProvider = <Config extends Record<string, unknown>>(
```

Comment:
Move `createConfigProvider` to it's own file?

This will export createConfigProvider from index.ts which we don't want as of now. The export was added just for testing the component.